### PR TITLE
Generate pdfFill function to return a buffer

### PR DESCRIPTION
**Pull request by aaschlote**

### CODE REVIEW

File: app/routes/shared/pdfDownloadLoader.ts [hunk 1]
```diff
@@ -46,7 +47,11 @@ export async function pdfDownloadLoader({ request }: LoaderFunctionArgs) {
...
   if (_.isEmpty(userData)) return redirect(flowId);
 
-  return new Response(await (await pdfFunction(userData)).save(), {
-    headers: createHeaders(filenameFunction()),
+  const fileContent = await pdfFunction(userData);
+  const fileSize = fileContent.length;
+  const filename = filenameFunction();
+
+  return new Response(fileContent, {
+    headers: createHeaders(filename, fileSize),
```
Comment (chohner) [inline app/routes/shared/pdfDownloadLoader.ts:55]: nice job :)

File: app/services/pdf/beratungshilfe/index.ts [hunk 2]
```diff
@@ -67,5 +67,5 @@ export async function beratungshilfePdfFromUserdata(
     filledPdf,
     await pdfFromReact(Handout(userData, "Merkblatt")),
   );
-  return filledPdf;
+  return await filledPdf.save();
```
Comment (chohner) [inline app/services/pdf/beratungshilfe/index.ts:70]: without known your constraints, this looks like a API change for the worse:
- tests are more complicated
- tests are slower
- concerns are more mixed

Could you motivate why the call-site shouldn't be responsible for further processing the returned `PdfDocument`?

File: app/routes/shared/pdfDownloadLoader.ts (hunk 1 above)
Comment (aaschlote) [reply app/routes/shared/pdfDownloadLoader.ts:55]: kudos to @pgurusinga he added it, I just copied&pasta 😆

File: app/routes/shared/pdfDownloadLoader.ts [hunk 3]
```diff
@@ -29,6 +29,7 @@ export function createHeaders(filename: string) {
   return {
     "Content-Type": "application/pdf",
     "Content-Disposition": `inline; filename=${encodeURIComponent(filename)}`,
+    "Content-Length": fileContentLength.toString(), // Add content length
```
Comment (chohner) [inline app/routes/shared/pdfDownloadLoader.ts:32]: remove the comment?

Comment (chohner) [reply app/routes/shared/pdfDownloadLoader.ts:32]: also - add a test for the header property? 😇

File: app/services/pdf/beratungshilfe/index.ts (hunk 2 above)
Comment (chohner) [reply app/services/pdf/beratungshilfe/index.ts:70]: aha! ok that makes sense. Could we wrap the existing functions in a `pdfDocToArrayBuffer()` function, so we can keep the tests untouched?

Comment (aaschlote) [reply app/services/pdf/beratungshilfe/index.ts:70]: do you mean do something like this:

```
pdfFunction: (userData) => pdfDocToArrayBuffer(beratungshilfePdfFromUserdata(userData)
```

**Review by chohner — APPROVED** [review]
nice

File: app/services/pdf/beratungshilfe/index.ts (hunk 2 above)
Comment (chohner) [reply app/services/pdf/beratungshilfe/index.ts]: yes. although looking at the function we could almost also just do it inline 😅
you call

File: app/routes/shared/pdfDownloadLoader.ts [hunk 4]
```diff
@@ -46,7 +41,11 @@ export async function pdfDownloadLoader({ request }: LoaderFunctionArgs) {
...
   if (_.isEmpty(userData)) return redirect(flowId);
 
-  return new Response(await (await pdfFunction(userData)).save(), {
-    headers: createHeaders(filenameFunction()),
+  const fileContent = await pdfFunction(userData);
+  const fileSize = fileContent.length;
+  const filename = filenameFunction();
+
+  return new Response(fileContent, {
+    headers: createPdfResponseHeaders(filename, fileSize),
```
Comment (pgurusinga) [inline app/routes/shared/pdfDownloadLoader.ts:49]: much better naming 👍

**Review by pgurusinga — APPROVED** [review]

**Pull request merged**